### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by corny**
This might break external code using `nl.NewRtAttrChild()`.

### CODE REVIEW

**Comment by aboch** [comment]
>This might break external code using nl.NewRtAttrChild().

Yes this is a problem as we started versioning this library.

I suggest to retain the `NewRtAttrChild()` definition along with your newly added `func (a *RtAttr) AddRtAttr()`

Please also note in this project one commit per PR is encouraged.
The NewRtAttr() change deserve its own PR.

Thanks.

**Comment by corny** [comment]
Thanks for your feedback. I've split up my refactoring into three pull requests.

File: link_linux.go [hunk 1]
```diff
@@ -1524,7 +1524,8 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 				if err != nil {
 					return nil, err
 				}
-				base.Protinfo = parseProtinfo(attrs)
+				protinfo := parseProtinfo(attrs)
+				base.Protinfo = &protinfo
```
Comment (aboch) [inline link_linux.go]: Why this change ?

**Comment by aboch** [comment]
LGTM

File: link_linux.go (hunk 1 above)
Comment (corny) [reply link_linux.go]: Yes .. it does not return a pointer any more.
